runtime/lua/vim/lsp/buf.lua

```diff
@@ -66,22 +66,31 @@ end
 --- Jumps to the declaration of the symbol under the cursor.
 ---@note Many servers do not implement this method. Generally, see |vim.lsp.buf.definition()| instead.
 ---
-function M.declaration()
+---@param options table|nil additional options
+---     - reuse_win: (boolean) Jump to existing window if buffer is already open.
+function M.declaration(options)
   local params = util.make_position_params()
+  params.options = options or {}
```

Comment:
I'm not sure if it's a good idea to add this to the payload that's also sent to the language server.  Worst case the language server crashes or responds with a failure due to unexpected properties.

I think a cleaner way would be to retrieve the handler and inject the option/ work with a closure